### PR TITLE
fix(compiler): support std json in struct fields

### PR DESCRIPTION
**Pull request by hasanaburayyan**
Moves the replacement of STD lib types to be more encompassing rather than just on function signatures. Also adds replacement of std types from symbol lookups to builtin if needed.

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

**Review by MarkMcCulloh — COMMENTED** [review]
Needs a test, otherwise looks good

**Comment by hasanaburayyan** [comment]
> Needs a test, otherwise looks good

@MarkMcCulloh I cant think of a good way to add a test without code from https://github.com/winglang/wing/pull/1285 OR just adding a dummy interface to core lib in the sdk. 

Actually this introduces a problem with also our container types. I think @staycoolcall911 is facing this potentially as well. If we define an interface in the SDK like:

```ts 
export interface Stuff {
	x: ImmutableArray,
	y: Json
}
```

and we have some wingcode like
```js
bring cloud;

inflight (msg:str): cloud.Stuff {
	return cloud.Stuff {
		x: ["a", "b"],
		y: Json "Cool"
	};
};
```

Today we will get:
Expected type to be "ImmutableArray", but got "Array<str>" instead
Expected type to be "Json", but got "Json" instead

My PR here only fixes "Json" to "Json" mappings. 

So basically I think this is a temp fix for a problem that needs a more holistic approached solution. @Chriscbr @MarkMcCulloh maybe we can take some time to huddle and chat about this tomorrow? 

In the interim I know this unblocks https://github.com/winglang/wing/pull/1285 so Im happy to let this code be included in that PR instead whatever we would prefer.

**Comment by hasanaburayyan** [comment]
Closing this PR in favor for better complete solution: https://github.com/winglang/wing/pull/1741
